### PR TITLE
General jumbo frame support/Network Refactor

### PR DESCRIPTION
**Pull request by 65278**
*Description of changes:*
We have jumbo frame support via virtio/proxmox, but no support for other device types like e1000e (etc.pp.) or software defined networking.
This PR adds general case jumbo frame support via netplan. To achieve this, I've refactored the networking code a little to unify software defined network devices with additional network devices.
While doing this, I removed the limitation of mutually exclusive DHCP and static IP. It's nonsensical and not in netplan.
I also gave all routes a metric and made the metric of default network 100 and other network devices 200. This is still buggy, but less buggy (and dangerous) than having all interfaces add default routes with a metric of 0. This only worked because netplan applies rules in order and was subject to breaking any time.

In a nutshell, this part of netplan https://netplan.readthedocs.io/en/stable/netplan-yaml/#properties-for-all-device-types is now unified in our API and our templating, making the adding of more features much easier. The end goal is to unify AdditionalNetworkDevices with DefaultNetworkDevice, but that's an API breaking change, so I plan on doing that in v1alpha2.

I've also fixed a couple of bugs on the way.

*Testing performed:*
Testing is still WIP

### CODE REVIEW

File: api/v1alpha1/proxmoxmachine_types.go [hunk 1]
```diff
@@ -231,6 +231,26 @@ type InterfaceConfig struct {
...
+}
+
+// Routing is shared fields across devices and VRFs
+type Routing struct {
+	// Routes are the routes associated with this interface.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	Routes []RouteSpec `json:"routes,omitempty"`
+
+	// RoutingPolicy is interface specific policy inserted into FiB (forwarding information base).
```
Comment (wikkyk) [inline api/v1alpha1/proxmoxmachine_types.go:250]:
```suggestion
	// RoutingPolicy is an interface-specific policy inserted into FIB (forwarding information base).
```

**Review by wikkyk — COMMENTED** [review]
Nitpick: FIB is variously referred to as FIB, FiB, fib. Please pick one and use it throughout for the sake of consistency.

**Review by wikkyk — APPROVED** [review]

File: docs/Troubleshooting.md [hunk 2]
```diff
@@ -67,3 +67,40 @@ If your capi-controller is too new, you can pass a `--core cluster-api:v1.6.1` d
 ## Calico fails in IPVS mode with loadBalancers to expose services
 Calico unfortunately does not test connectivity when it choses a node ip to use for IPVS communication.
 This can be altered manually. More on this topic in [Calicos documentation](https://docs.tigera.io/calico/latest/networking/ipam/ip-autodetection#autodetection-methods).
+
+## Nodes fail to deploy/have wrong node-ip with mixed interface models
+Kubelet chooses the first interface to acquire a node-ip for kubeadm. The first
+interface is defined by the in kernel order, which is defined by the order the
```
Comment (wikkyk) [inline docs/Troubleshooting.md:73]: in-kernel

**Review by wikkyk — APPROVED** [review]

**Comment by wikkyk** [comment]
checks and e2e passed, merging

**Review by wikkyk — APPROVED** [review]
